packages/neon/neon_talk/lib/src/dialogs/create_room.dart

```diff
@@ -0,0 +1,190 @@
+class TalkCreateRoomDialog extends StatefulWidget {
+  /// Creates a new Talk create room dialog.
+  const TalkCreateRoomDialog({
+    super.key,
+  });
+
+  @override
+  State<TalkCreateRoomDialog> createState() => _TalkCreateRoomDialogState();
+}
+
+class _TalkCreateRoomDialogState extends State<TalkCreateRoomDialog> {
+  late final values = {
+    spreed.RoomType.oneToOne: TalkLocalizations.of(context).roomTypeOneToOne,
+    spreed.RoomType.group: TalkLocalizations.of(context).roomTypeGroup,
+    spreed.RoomType.public: TalkLocalizations.of(context).roomTypePublic,
+  };
+
+  final formKey = GlobalKey<FormState>();
+  final controller = TextEditingController();
+  final focusNode = FocusNode();
+
+  spreed.RoomType? selectedType;
+  core.AutocompleteResult? selectedResult;
+
+  void changeType(spreed.RoomType? type) {
+    controller.clear();
+
+    setState(() {
+      selectedResult = null;
+      selectedType = type;
+    });
+  }
+
+  void submit() {
+    if (formKey.currentState!.validate()) {
+      final type = selectedType!;
+      final isPublic = type == spreed.RoomType.public;
+
+      Navigator.of(context).pop<TalkCreateRoomDetails>(
+        (
+          type: type,
+          roomName: isPublic ? controller.text : null,
+          invite: !isPublic ? selectedResult : null,
+        ),
+      );
+    }
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    focusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final body = Column(
+      crossAxisAlignment: CrossAxisAlignment.end,
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        for (final type in values.keys)
+          ListTile(
+            title: Text(values[type]!),
+            leading: Icon(
+              // coverage:ignore-start
+              switch (type) {
+                // coverage:ignore-end
+                spreed.RoomType.oneToOne => AdaptiveIcons.person,
+                spreed.RoomType.group => AdaptiveIcons.group,
+                spreed.RoomType.public => AdaptiveIcons.link,
+                _ => throw UnimplementedError('$type is not a valid type'), // coverage:ignore-line
+              },
+            ),
+            trailing: Radio(
+              value: type,
+              groupValue: selectedType,
+              onChanged: changeType,
+            ),
+            onTap: () {
+              changeType(type);
+            },
+          ),
+        const Divider(),
+        switch (selectedType) {
+          spreed.RoomType.oneToOne || spreed.RoomType.group => buildAutocomplete(),
+          spreed.RoomType.public => TextFormField(
+              controller: controller,
+              focusNode: focusNode,
+              validator: (input) => validateNotEmpty(context, input),
+              decoration: InputDecoration(
+                hintText: TalkLocalizations.of(context).roomCreateRoomName,
+              ),
+              autofocus: true,
```

Comment:
The autofocus makes the textfield flicker when selecting another one.
I don't know if we can solve this with some focusNode magic or not but it would be great.